### ISSUE TITLE
Consistent handling of evmc_result

### ISSUE THREAD

**Issue by axic** [ethereum/evmc#87]
1. `evmc_execute_fn` returns it as value
2. `evmc_call_fn` expects a user specified pointer refering to it, to be filled
3. `evmc_release_result_fn` expects a pointer to it, yet it 

I assume 3. because you wanted it to behave like a C++ destructor.

It would be nice to:
- `evmc_call_fn` to return the result struct as a value
- `evmc_release_result_fn` not take a pointer, but a struct

Both these make it even more clear that `evmc_result` struct cannot be extended by the VM.

**Comment by chfast** [comment]
I had problems with returning structs with unions in Python so now every host function returning a struct mirrors C ABI. This is actually not so bad because now you can return an error like e.g. in `evmc_blockhash()`.

For `evmc_call_fn` this could be a good change.
For `evmc_release_result_fn` it should be a pointer not to copy the whole struct.

**Comment by axic** [comment]
> For evmc_release_result_fn it should be a pointer not to copy the whole struct.

Good point, can you document this reason? Too bad C doesn't have reference syntax, would make this nicer.

Now I wonder if returning a pointer in all cases is better and let `release` handle it. That we could discuss as a separate thread.
